src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -839,11 +839,33 @@ class ParseLiveQueryServer {
       return;
     }
 
+    const subscription = subscriptionInfo.subscription;
+    const className = subscription.className;
+    const trigger = getTrigger(className, 'beforeUnsubscribe', Parse.applicationId);
+    if (trigger) {
+      const auth = await this.getAuthFromClient(client, request.requestId, request.sessionToken);
+      if (auth && auth.user) {
+        request.user = auth.user;
+      }
+
+      request.sessionToken = subscriptionInfo.sessionToken;
+      request.useMasterKey = client.hasMasterKey;
+      request.installationId = client.installationId;
+
+      try {
+        await runTrigger(trigger, `beforeUnsubscribe.${className}`, request, auth);
+      } catch (error) {
+        Client.pushError(parseWebsocket, error.code || 141, error.message || error, false);
```

Comment:
No hard coded error numbers

Reply:
Lol, it was copied from a few lines above 🙃